tests/integration/mpitests/mpitest_parallel_archipelago_and_multiprocessing.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+
+from bingo.evolutionary_optimizers.parallel_archipelago import \
+    ParallelArchipelago
+from bingo.evolutionary_algorithms.mu_plus_lambda import MuPlusLambda
+from bingo.evolutionary_optimizers.island import Island
+from bingo.selection.tournament import Tournament
+from bingo.chromosomes.multiple_values import SinglePointCrossover, \
+    SinglePointMutation, MultipleValueChromosomeGenerator
+from bingo.evaluation.fitness_function import FitnessFunction
+from bingo.evaluation.evaluation import Evaluation
+
+from mpitest_util import mpi_assert_true, run_t_in_module
+
+from mpi4py import MPI
+COMM = MPI.COMM_WORLD
+COMM_RANK = COMM.Get_rank()
+COMM_SIZE = COMM.Get_size()
+
+POPULATION_SIZE = 100
+OFFSPRING_SIZE = 100
+N_PROC = 2
+
+
+class MagnitudeFitness(FitnessFunction):
+    def __call__(self, individual):
+        self.eval_count += 1
+        return np.linalg.norm(individual.values)
+
+
+def evaluation():
+    return Evaluation(MagnitudeFitness(), multiprocess=N_PROC)
+
+
+def ea():
+    selection = Tournament(5)
+    crossover = SinglePointCrossover()
+    mutation = SinglePointMutation(np.random.random)
+    return MuPlusLambda(evaluation(), selection, crossover,
+                        mutation, 0.4, 0.4, OFFSPRING_SIZE)
+
+
+def generator():
+    return MultipleValueChromosomeGenerator(np.random.random, 5)
+
+
+def multi_process_parallel_archipelago():
+    island = Island(ea(), generator(), POPULATION_SIZE)
+    # TODO: is it intended that non_blocking steps by sync_frequency?
+    #       what if n_steps < sync_frequency?
+    #       also < vs. <= in avg_age vs. target_age
```

Comment:
Some questions about using the parallel archipelago with non_blocking

Reply:
yep, it is intended to step by sync_frequency.

we have actually run into the issue where n_steps < sync_frequency,  it runs syn_frequency steps.  This is not ideal and I'm open to a fix here.

should sync_frequency have a default of None, where we define it based on n_steps?  then allow an override when it is specified explicitly?


in your last question I'm guessing you're referring to this code:
```python
while average_age < target_age:
            self.island.evolve(self._sync_frequency,
...
```
to me it makes sense to stop evolving when the average age is equal to the target age.  (I think i might be missing the question here though)

Reply:
Having sync_frequency default to n_steps with the option to override it sounds reasonable to me.

Yeah I was asking about that code snippet. I was finding a case where the average_age was greater than the target_age but that was due to the sync_frequency. Using `average_age < target_age` makes sense to me now.

Reply:
Decided on changing `sync_frequency` to 1 if `n_steps < sync_frequency`, otherwise keep `sync_frequency` at 10 (or whatever the user set).